rllib/algorithms/algorithm_config.py

```diff
@@ -2665,25 +2670,52 @@ def get_default_rl_trainer_class(self) -> Union[Type["RLTrainer"], str]:
         """
         raise NotImplementedError
 
-    def get_trainer_runner_config(
-        self, module_spec: Optional[ModuleSpec] = None
-    ) -> TrainerRunnerConfig:
+    def get_marl_module_spec(
+        self,
+        policy_dict: Dict[str, PolicySpec],
+        policies_to_train: Callable[[PolicyID, SampleBatchType], bool],
+    ) -> MultiAgentRLModuleSpec:
+        """Returns the MultiAgentRLModule spec based on the given policy spec dict.
 
-        if module_spec is None:
-            module_spec = SingleAgentRLModuleSpec()
+        Args:
+            policy_dict: The policy spec dict. Using this dict, we can determine the
+                inferred values for observation_space, action_space, and config for
+                each policy. If the module spec does not have these values specified,
+                they will get auto-filled with these values obtrained from the policy
+                spec dict. Here we are relying on the policy's logic for infering these
+                values from other sources of information (e.g. environement)
+            policies_to_train: The policies to train. This can be optionally used to
+                construct the MultiAgentRLModuleSpec (if necessary).
+        """
 
-        if isinstance(module_spec, SingleAgentRLModuleSpec):
-            if module_spec.module_class is None:
-                module_spec.module_class = self.rl_module_class
+        marl_module_spec = self.rl_module_spec
 
-            if module_spec.observation_space is None:
-                module_spec.observation_space = self.observation_space
+        # If the module is single-agent convert it to multi-agent spec
+        if isinstance(marl_module_spec, SingleAgentRLModuleSpec):
+            marl_module_spec = MultiAgentRLModuleSpec(
+                module_class=MultiAgentRLModule,
+                module_specs={k: marl_module_spec for k in policy_dict.keys()},
+            )
 
-            if module_spec.action_space is None:
-                module_spec.action_space = self.action_space
+        # Make sure that policy_dict and marl_module_spec have similar keys
+        if set(policy_dict.keys()) != set(marl_module_spec.module_specs.keys()):
+            raise ValueError("Policy dict and module spec have different keys!")
 
+        # fill in the missing values from the module spec
+        for module_id in policy_dict:
+            policy_spec = policy_dict[module_id]
+            module_spec = marl_module_spec.module_specs[module_id]
+
+            if module_spec.observation_space is None:
+                module_spec.observation_space = policy_spec.observation_space
+            if module_spec.action_space is None:
+                module_spec.action_space = policy_spec.action_space
             if module_spec.model_config is None:
-                module_spec.model_config = self.model
+                module_spec.model_config = policy_spec.config.get("model", {})
```

Comment:
is this block temporary? it's just bridging between the RLModule and Policy worlds right?
if so, can we add a TODO/Note?

Reply:
Yes, it's temporary until policy co-exists with the RLModule / Learner API. Once we re-write sampler / rollout workers to drop policy, then we won't need this method anymore. instead of creating policy_dicts we will directly create marl_module_specs.

Reply:
added TODO and Note.